### ISSUE TITLE
JSON parse error when displaying validation error in tree mode

### ISSUE THREAD

**Issue by powerslave** [josdejong/jsoneditor#314]
Using schema and json provided below, I get a JSON parse error, and a broken tree view.
`Failed to parse JSON schema: SyntaxError: Unexpected token ' in JSON at position 0`

This schema/json combo is supposed to fail validation. I verified proper failing validation with both http://jsonschemalint.com/draft4/ and http://www.jsonschemavalidator.net/ The failing part in json is `"outside":{}`

* I can repro this both on `jsononlineeditor.org` and a local app that loads `cdnjs/5.5.6/jsoneditor.min.js`
* Issue seems to happen when when _displaying_ the validation error, validation per se works fine.
* Only happens in tree view. Validation error display in code view is correct.

##### reproduction:

1: set this schema: http://www.jsoneditoronline.org/?id=92b1743f2e0d71f85b406818f0538841

```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "definitions": {
    "schemaArray": {
      "type": "array",
      "items": {
        "$ref": "#"
      }
    }
  },
  "type": "object",
  "properties": {
    "$ref": {
      "type": "string"
    },    
    "additionalProperties": {},
    "definitions": {
      "type": "object",
      "additionalProperties": {
        "$ref": "#"
      }
    },
    "properties": {
      "type": "object",
      "additionalProperties": {
        "$ref": "#"
      }
    },
    "type": {}
  },
  "additionalProperties": false
}
```

----------------

2: and load this json: http://www.jsoneditoronline.org/?id=6a1361edd42146d7c93c0cb9f746f2b8

```json
{
  "type": "object",
  "properties": {
    "Sub": {
      "type": "object",
      "$ref": "#/definitions/sub.json"
    }
  },
  "definitions": {
    "sub.json": {
      "type": "object",
      "properties": {},
      "additionalProperties": false,
      "outside":{}
    }
  }
}
```

Happy to help with more details

**Comment by josdejong** [comment]
Thanks for reporting, this issue originates with JSONEditor not yet understanding `$ref` in a schema.

See also #302 and #310

**Comment by powerslave** [comment]
thanks @josdejong 
FYI: In one of my scenarios I could workaround by adding `onEditable: noEditing`, keeping validation alive in tree view. In the other scenario `noEditing` did not help, have to live with validation in text view only for now.

**Comment by josdejong** [comment]
That makes sense, thanks for the feedback.

**Comment by rsheptolut** [comment]
@josdejong Hi! Love the JSON editor! Any news on when it will fully support "$ref"s?

**Comment by rsheptolut** [comment]
@josdejong also, maybe it deserves its own issue, but here's the same error in the same cicrumstances not involving `$ref`, but involving the much needed `patternProperties` json schema property. I intentionally tried to simplify the hell out of it.

### json
```json
{
  "Test": 14,
  "Test1": "sd"
}
```

### schema
```json
{
  "type": "object",
  "patternProperties": {
    "Test.*" : {
      "type": "number"
    }
  }
}
```

This gives the same error as @powerslave reported back in july: `Failed to parse JSON schema: SyntaxError: Unexpected token ' in JSON at position 0`

**Comment by josdejong** [comment]
Thanks @rsheptolut, I haven't picked that up yet. Help is welcome. 

I can reproduce the issue you're explaining, thanks. Not yet sure whether it has a similar cause or is completely unrelated.

**Comment by josdejong** [comment]
This should have been fixed in v5.5.8 thanks to Stevens fix (#326).
